core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeExpressionBuilder.scala

```diff
@@ -348,6 +351,20 @@ class SnowflakeExpressionBuilder()
     val characters = Option(ctx.string()).map(_.accept(this)).toList
     functionBuilder.buildFunction(ctx.trim.getText, expression :: characters)
   }
+
+  override def visitBuiltinExtract(ctx: BuiltinExtractContext): ir.Expression = {
+    val part = ir.Id(removeQuotes(ctx.part.getText))
+    val date = ctx.expr().accept(this)
+    functionBuilder.buildFunction(ctx.EXTRACT().getText, Seq(part, date))
+  }
+
+  override def visitBuiltinArrayAgg(ctx: BuiltinArrayAggContext): ir.Expression = {
```

Comment:
By folding WITHIN GROUP in to [the expression](https://github.com/databrickslabs/remorph/blob/main/core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala#L377), then this visitor can be deleted along with the rule.